### ISSUE TITLE
Eslint inline rule config applies for code above the config

### ISSUE THREAD

**Issue by kkmuffme** [eslint/eslint#16069]
### Environment

Node version: v16.14.2
npm version: v8.12.1
Local ESLint version: Not found
Global ESLint version: v8.17.0 (Currently used)
Operating System: linux 4.18.0-348.7.1.el8_5.x86_64

### What parser are you using?

Default (Espree)

### What did you do?

```js
const a = 'a';

/* eslint quotes: ["error", "double"] */
const b = 'b';
```

### What did you expect to happen?

"quotes" error only for `const b = 'b';`

When changing/setting a eslint rule config, the change should only apply for code below that line.

### What actually happened?

Got an error for code above the rule too.

### Additional comments

In reality it's not about the quotes comment, but about things like eslint max-nested-callbacks. Where one sometimes need to increase the number of callbacks allowed by 1 in some cases.
This is currently not possible, leading to overall less strictly checked code. e.g. I use max-nested-callbacks 2, but in some rare cases I need 3. Using eslint-disable-next-line is not an option, since I would need to put that comment 100s of lines and also I don't want to disable it, but just set the config level to allow 1 more nested callback.

**Comment by nzakas** [comment]
This is working as intended. Rule configs inside of files apply to the entire file. You can only disable rules around specific areas, you can’t configure them this way.
